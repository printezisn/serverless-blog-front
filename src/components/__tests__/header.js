import React from "react"
import renderer from "react-test-renderer"

import Header from "../header"

jest.mock("../menu", () => "Menu")

describe("Header", () => {
    const oldEnvVariables = process.env

    afterEach(() => {
        process.env.GATSBY_EDNABLOG_SITE_TITLE =
            oldEnvVariables.GATSBY_EDNABLOG_SITE_TITLE
        process.env.GATSBY_EDNABLOG_SITE_SUBTITLE =
            oldEnvVariables.GATSBY_EDNABLOG_SITE_SUBTITLE
    })

    it("renders correctly", () => {
        process.env.GATSBY_EDNABLOG_SITE_TITLE = "Edna Blog"
        process.env.GATSBY_EDNABLOG_SITE_SUBTITLE =
            "(E)very (D)eveloper (N)eeds (A) blog"

        const header = renderer.create(<Header />).toJSON()
        expect(header).toMatchSnapshot()
    })
})
