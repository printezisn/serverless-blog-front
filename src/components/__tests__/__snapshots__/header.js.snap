// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <Menu />
  <div
    className="banner"
  >
    <h1
      className="title is-1 has-text-centered"
    >
      <a
        href="/"
      >
        <i
          className="fas fa-laptop-code"
        />
        Edna Blog
      </a>
    </h1>
    <p
      className="subtitle has-text-centered"
    >
      (E)very (D)eveloper (N)eeds (A) blog
    </p>
  </div>
</header>
`;
